crates/router/src/core/payments/flows/authorize_flow.rs

```diff
@@ -95,37 +95,71 @@ impl Feature<api::Authorize, types::PaymentsAuthorizeData> for types::PaymentsAu
 
             metrics::PAYMENT_COUNT.add(&metrics::CONTEXT, 1, &[]); // Metrics
 
-            let save_payment_result = tokenization::save_payment_method(
-                state,
-                connector,
-                resp.to_owned(),
-                maybe_customer,
-                merchant_account,
-                self.request.payment_method_type,
-                key_store,
-            )
-            .await;
-
-            let pm_id = match save_payment_result {
-                Ok(payment_method_id) => Ok(payment_method_id),
-                Err(error) => {
-                    if resp.request.setup_mandate_details.clone().is_some() {
-                        Err(error)
-                    } else {
-                        logger::error!(save_payment_method_error=?error);
-                        Ok(None)
-                    }
-                }
-            }?;
-
-            Ok(mandate::mandate_procedure(
-                state,
-                resp,
-                maybe_customer,
-                pm_id,
-                connector.merchant_connector_id.clone(),
-            )
-            .await?)
+            if resp.request.setup_mandate_details.clone().is_some() {
+                let payment_method_id = tokenization::save_payment_method(
+                    state,
+                    connector,
+                    resp.to_owned(),
+                    maybe_customer,
+                    merchant_account,
+                    self.request.payment_method_type,
+                    key_store,
+                )
+                .await?;
+                Ok(mandate::mandate_procedure(
+                    state,
+                    resp,
+                    maybe_customer,
+                    payment_method_id,
+                    connector.merchant_connector_id.clone(),
+                )
+                .await?)
+            } else {
+                let arbiter = actix::Arbiter::try_current()
+                    .ok_or(errors::ApiErrorResponse::InternalServerError)
+                    .into_report()
+                    .attach_printable("arbiter retrieval failure")
+                    .map_err(|err| {
+                        logger::error!(?err);
+                        err
+                    })
+                    .ok();
```

Comment:
Can't we use tokio::task here?

Reply:
Had picked this up from webhooks. But recently webhooks has been changed to use tokio::spawn, Will do the same here as well.